manifests/repo.pp

```diff
@@ -118,23 +118,23 @@
   Enum['present', 'absent'] $ensure,
   String[1] $release,
   Mrepo::Arch $arch,
-  Hash[String, String] $urls                = {},
-  String[1] $metadata                       = 'repomd',
-  Mrepo::Update $update                     = 'nightly',
-  Variant[String[1], Integer] $hour         = '0',
-  Variant[String[1], Integer] $minute       = '0',
-  Optional[String[1]] $iso                  = undef,
-  String[1] $repotitle                      = $name,
-  Variant[String[1], Integer] $gen_timeout  = '1200',
-  Variant[String[1], Integer] $sync_timeout = '1200',
-  Enum['std', 'ncc', 'rhn'] $type           = 'std',
-  Optional[String[1]] $typerelease          = undef,
-  Optional[String[1]] $mrepo_env            = undef,
-  Stdlib::Absolutepath $mrepo_command       = '/usr/bin/mrepo',
-  String $mrepo_options                     = '-qgu',
-  Optional[String[1]] $mrepo_logging        = undef,
-  Optional[String[1]] $ncc_username         = undef,
-  Optional[String[1]] $ncc_password         = undef,
+  Hash[String, String] $urls                                                                            = {},
+  Variant[Enum['yum','apt','repomd','repoview'],Array[Enum['yum','apt','repomd','repoview']]] $metadata = 'repomd',
```

Comment:
Like with bellow parameter `update`, it will be clearer to create a custom data type.

Reply:
@mrolli - are you game to do the adjustment mentioned here? If not I am happy to do it.

Reply:
@genebean Hope this is what you meant. Didn't squash by intention to make the addition of the new datatype explicit.

Reply:
I'm pretty sure it's what he had in mind.  Merged!  Thanks for the fix.